github/github_test.go

```diff
@@ -705,7 +704,7 @@ func TestDo_nilContext(t *testing.T) {
 	req, _ := client.NewRequest("GET", ".", nil)
 	_, err := client.Do(nil, req, nil)
 
-	if !reflect.DeepEqual(err, errors.New("context must be non-nil")) {
+	if err.Error() != "context must be non-nil" {
```

Comment:
consider creating a 

```
var errNonNilContext = errors.New("context must be non-nil")
```
in github.go and then use `errors.Is()` here.

Reply:
Taking a look at: https://blog.golang.org/go1.13-errors ...

If we do this, would we need to change the README.md such that this repo would then require Go 1.13 instead of Go 1.9?

Does anyone care about Go 1.9 compatibility anymore?

Reply:
Since this repo supports go mods (which started on Go 1.11 if I'm not wrong) I think supporting Go 1.9 is...irrelevant?

Reply:
OK, it's actually completely unrelated, so I think I'll just revert this change and open a new issue for it.

Reply:
Here is the new issue: https://github.com/google/go-github/issues/1846

Reply:
Reverted this change.  Resolving conversation.